### PR TITLE
[llvm-lit][test] Resolved typo in raising InternalShellError for export command in lit's internal shell

### LINKED ISSUE #102386: [llvm-lit] InternalShellError in compiler-rt test using export command
A compiler-rt test using the `export` command is failing when run with lit’s internal shell.

Error message:

```
UNRESOLVED: AddressSanitizer-x86_64-linux :: TestCases/Posix/high-address-dereference.c (129 of 9570)
******************** TEST 'AddressSanitizer-x86_64-linux :: TestCases/Posix/high-address-dereference.c' FAILED ********************
Exception during script execution:
TypeError: InternalShellError.__init__() missing 1 required positional argument: 'message'
```

### PR DESCRIPTION
**Pull request by connieyzhu**
This patch fixes the incorrect usage of lit's built-in `export` command. There is a typo in raising the error itself where the error being raised had the wrong number of parameters passed in.

Fixes https://github.com/llvm/llvm-project/issues/102386.

### CODE REVIEW

File: llvm/utils/lit/lit/TestRunner.py [hunk 1]
```diff
@@ -356,7 +356,7 @@ def executeBuiltinPopd(cmd, shenv):
 def executeBuiltinExport(cmd, shenv):
     """executeBuiltinExport - Set an environment variable."""
     if len(cmd.args) != 2:
-        raise InternalShellError("'export' supports only one argument")
+        raise InternalShellError(cmd, "'export' supports only one argument")
```
Comment (ilovepi) [inline llvm/utils/lit/lit/TestRunner.py:359]: There should be an accompanying test.

File: compiler-rt/test/asan/TestCases/Posix/high-address-dereference.c [hunk 2]
```diff
@@ -5,12 +5,11 @@
...
-// RUN: not %run %t 0x0000000000000000 2>&1 | FileCheck %s --check-prefixes=ZERO,HINT-PAGE0
-// RUN: not %run %t 0x0000000000000FFF 2>&1 | FileCheck %s --check-prefixes=LOW1,HINT-PAGE0
-// RUN: not %run %t 0x0000000000001000 2>&1 | FileCheck %s --check-prefixes=LOW2,HINT-NONE
-// RUN: not %run %t 0x4141414141414141 2>&1 | FileCheck %s --check-prefixes=HIGH,HINT-HIGHADDR
-// RUN: not %run %t 0xFFFFFFFFFFFFFFFF 2>&1 | FileCheck %s --check-prefixes=MAX,HINT-HIGHADDR
+// RUN: %env_asan_opts=print_scariness=1 not %run %t 0x0000000000000000 2>&1 | FileCheck %s --check-prefixes=ZERO,HINT-PAGE0
+// RUN: %env_asan_opts=print_scariness=1 not %run %t 0x0000000000000FFF 2>&1 | FileCheck %s --check-prefixes=LOW1,HINT-PAGE0
+// RUN: %env_asan_opts=print_scariness=1 not %run %t 0x0000000000001000 2>&1 | FileCheck %s --check-prefixes=LOW2,HINT-NONE
+// RUN: %env_asan_opts=print_scariness=1 not %run %t 0x4141414141414141 2>&1 | FileCheck %s --check-prefixes=HIGH,HINT-HIGHADDR
+// RUN: %env_asan_opts=print_scariness=1 not %run %t 0xFFFFFFFFFFFFFFFF 2>&1 | FileCheck %s --check-prefixes=MAX,HINT-HIGHADDR
```
Comment (ilovepi) [inline compiler-rt/test/asan/TestCases/Posix/high-address-dereference.c:12]: This seems unrelated to this patch, is there a reason its included here?

**Review by ilovepi — CHANGES_REQUESTED** [review]
I'm fine w/ the code change in TestRunner.py, but 

1) this needs a test in lit's test suite
1) the compiler-rt change should be split out on its own

File: compiler-rt/test/asan/TestCases/Posix/high-address-dereference.c (hunk 2 above)
Comment (connieyzhu) [reply compiler-rt/test/asan/TestCases/Posix/high-address-dereference.c:12]: I included it because it was related to the RUN line that caused the InternalShellError. I'm guessing I should have put this change in a separate patch?

File: llvm/utils/lit/lit/TestRunner.py (hunk 1 above)
Comment (connieyzhu) [reply llvm/utils/lit/lit/TestRunner.py:359]: I created the test in my latest change. I only made a test to show the failing behavior that is used here, but I wasn't sure if I should make tests to show that the export command works. There are other lit tests that use the export command, which do show to some extent that the command works. Do you think that's enough, or should I add more tests?

**Review by ilovepi — APPROVED** [review]
LGTM.

File: llvm/utils/lit/lit/TestRunner.py (hunk 1 above)
Comment (ilovepi) [reply llvm/utils/lit/lit/TestRunner.py:359]: Your change literally only affects the error message, so I don't think that's required. As long as other tests are passing/failing the same as before your patch then I don't think you need to update the world. Now, it would be nice to improve the state of testing, but that shouldn't block you from making the error consistent.
